pkg/scheduler/objects/preemption.go

```diff
@@ -505,7 +507,7 @@ func (p *Preemptor) tryNodes() (string, string, []*Allocation, bool) {
 	// call predicates to evaluate each node
 	result := p.checkPreemptionPredicates(predicateChecks, victimsByNode)
 	if result != nil && result.success {
-		return result.nodeID, result.instType, result.victims, true
+		return result.nodeID, p.nodeInstanceTypeMap[result.nodeID], result.victims, true
```

Comment:
Always check the result of a map lookup, just in case. If it's not found, log a BUG message (see elsewhere for examples).

Reply:
Use unknown if not found.